content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java

```diff
@@ -161,10 +166,120 @@ public void init() {
 
 		// Populate content upload headers used in uploadExternalContent
 		CONTENT_UPLOAD_HEADERS.putAll(BASE_HEADERS);
 		CONTENT_UPLOAD_HEADERS.put(HEADER_CONTENT, CONTENT_TYPE_BINARY);
+
+		try {
+			// Get the webhook url
+			String webhookUrl = getWebhookUrl(Optional.empty());
+			boolean webhooksSetup = false;
+			// Check to see if any webhooks have already been set up for this url
+			for (Webhook webhook : getWebhooks()) {
+				if (StringUtils.isNotEmpty(webhook.getUrl()) && webhook.getUrl().equals(webhookUrl)) {
+					webhooksSetup = true;
+					break;
+				}
+			}
+
+			if (!webhooksSetup) {
+				// No webhook set up for this url, set one up
+				setupWebhook(webhookUrl);
+			}
+		} catch (Exception e) {
+			log.error(e.getLocalizedMessage(), e);
+		}
+	}
+
+	public String setupWebhook(String webhookUrl) throws Exception {
+		String id;
+
+		Map<String, Object> data = new HashMap<String, Object>();
+
+		List<String> types = new ArrayList<>();
+		types.add("SIMILARITY_COMPLETE");
+		types.add("SUBMISSION_COMPLETE");
+
+		data.put("signing_secret", apiKey);
+		data.put("url", webhookUrl);
+		data.put("description", "Sakai " + sakaiVersion);
+		data.put("allow_insecure", false);
+		data.put("event_types", types);
+
+		HashMap<String, Object> response = makeHttpCall("POST",
+				getNormalizedServiceUrl() + "webhooks",
+				WEBHOOK_SETUP_HEADERS,
+				data,
+				null);
+
+		// Get response:
+		int responseCode = !response.containsKey(RESPONSE_CODE) ? 0 : (int) response.get(RESPONSE_CODE);
+		String responseMessage = !response.containsKey(RESPONSE_MESSAGE) ? "" : (String) response.get(RESPONSE_MESSAGE);
+		String responseBody = !response.containsKey(RESPONSE_BODY) ? "" : (String) response.get(RESPONSE_BODY);
+
+		if ((responseCode >= 200) && (responseCode < 300) && (responseBody != null)) {
+			// create JSONObject from responseBody
+			JSONObject responseJSON = JSONObject.fromObject(responseBody);
+			if (responseJSON.has("id")) {
+				id = responseJSON.getString("id");
+			} else {
+				throw new Error("Error setting up webhook - returned with no ID");
+			}
+		} else {
+			throw new Error(responseMessage);
+		}
+
+		return id;
+	}
+	
+	public ArrayList<Webhook> getWebhooks() throws Exception {
+		ArrayList<Webhook> webhooks = new ArrayList<>();
+
+		HashMap<String, Object> response = makeHttpCall("GET",
+				getNormalizedServiceUrl() + "webhooks",
+				BASE_HEADERS,
+				null,
+				null);
+
+		// Get response:
+		int responseCode = !response.containsKey(RESPONSE_CODE) ? 0 : (int) response.get(RESPONSE_CODE);
+		String responseMessage = !response.containsKey(RESPONSE_MESSAGE) ? "" : (String) response.get(RESPONSE_MESSAGE);
+		String responseBody = !response.containsKey(RESPONSE_BODY) ? "" : (String) response.get(RESPONSE_BODY);
+
+		if ((responseCode < 200) || (responseCode >= 300) || (responseBody == null)) {
+			throw new Error(responseMessage);
+		} else if (responseBody != "" && responseBody != "[]") {
```

Comment:
Don't use "=" when comparing strings in Java
else if(StringUtils.isNotEmpty(responseBody) && !"[]".equals(responseBody))